llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -32308,6 +32308,54 @@ bool X86TargetLowering::isInlineAsmTargetBranch(
   return Inst.equals_insensitive("call") || Inst.equals_insensitive("jmp");
 }
 
+static SDValue getFlagsOfCmpZeroFori1(SelectionDAG &DAG, const SDLoc &DL,
+                                      SDValue Mask) {
+  EVT Ty = MVT::i8;
+  auto V = DAG.getBitcast(MVT::i1, Mask);
+  auto VE = DAG.getZExtOrTrunc(V, DL, Ty);
+  auto Zero = DAG.getConstant(0, DL, Ty);
+  SDVTList X86SubVTs = DAG.getVTList(Ty, MVT::i32);
+  auto CmpZero = DAG.getNode(X86ISD::SUB, DL, X86SubVTs, Zero, VE);
```

Comment:
why `SUB` as opposed to `CMP`?

Reply:
Same reason I mentioned in an earlier PR: for machine CSE, we always emit SUB instead of CMP when building node.

Reply:
IIUC, you are encoding -flags in the mask? If so, this needs a comment/explination.

Reply:
No. I am generating the condition code COND_NE for CLOAD/CSTORE.

Reply:
Then my not cmp?

Reply:
SUB instruction can set COND_E/COND_NE itself. No need an extra NOT instruction.

Reply:
CMP/TEST can't see both of them aswell?

Reply:
For machine CSE cases like `a ? -a: b;`, SUB will be transformed to TEST if possible, which can verified from test cf.ll